compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -172,6 +172,12 @@ public class Types {
     private boolean ignoreObjectTypeIds = false;
     private static final String BASE_16 = "base16";
 
+    private static final BigDecimal DECIMAL_MAX =
+            new BigDecimal("9.999999999999999999999999999999999e6144", MathContext.DECIMAL128);
+
```

Comment:
Is this value correct?
According to the pdf that you shared,
```
The finite numbers are defined by a sign, an exponent (which is a power of ten), and a 
decimal integer coefficient. The value of a finite number is given by (–1)sign × coefficient × 10^exponent. 
```
According to this coefficient is a decimal integer. So `decimal_max` value should be `9999999999999999999999999999999999e6144`. Isn't it?

Reply:
Hope you have read the pointed documentation as mentioned in https://github.com/ballerina-platform/ballerina-lang/pull/37823#issuecomment-1265156320, at the end, it is mentioned that `For the decimal32 format, the largest normal number is 9.999999 × 10Emax` 
If `9999999999999999999999999999999999e6144` is max value for decimal128, how does 6144 becomes max value?

Also please go through the topic `decContext module` at https://speleotrove.com/decimal/decnumber.pdf

The sentence you pointed out says how decimal value is represented in the implementation.